Set the redirect URL back to the editor when users finish the plan purchase

alshakero opened the pull request:
This PR sets the block editor URL as a post-checkout redirect URL. It doesn't however try to fix the stale cookie problem explained in pbOQVh-lb-p2.

#### Testing instructions

* Enable Sandbox.
* Run `yarn dev --sync` on `apps/wpcom-block-editor`.
* Open block editor of the free site with the feature flag `?flags=post-editor/checkout-overlay`, e.g. `calypso.localhost:3000/block-editor/post/your_sandboxed_free_site.wordpress.com?flags=post-editor/checkout-overlay`
* Open up developer tools > console, and run the following code:

```js
let cartData = { products: [{
    product_id: 1009,
    product_slug: 'personal-bundle',
}] };
window.wp.hooks.doAction('a8c.wpcom-block-editor.openCheckoutModal', cartData);
```

* Finish the payment process
* You should be redirected back to the editor.

Fixed: #46362 and supersedes https://github.com/Automattic/wp-calypso/pull/45902

sirbrillig commented on client/my-sites/checkout/composite-checkout/use-get-thank-you-url.js:
```diff
@@ -102,6 +103,13 @@ export function getThankYouPageUrl( {
+	// If the user is making a purchase/upgrading within the editor,
+	// we want to return them back to the editor after the purchase is successful.
+	if ( isInEditor && ! hasEcommercePlan( cart ) ) {
+		saveUrlToCookie( window?.location.href );
```
If possible, it would be great to add a test for this behavior in `client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js`

sirbrillig reviewed (commented):
I haven't run through the tests, but overall this looks good to me from a code perspective (with a few requests I made inline)!

tjcafferkey commented:
I haven't **yet** tested this but I think there might be an issue with this scenario:

* From the Editor, open checkout
* Checkout with PayPal
* Accept purchase of Quick Start Concierge Session
* Also purchase this with PayPal
* Check redirect destination link, I think it could be broken.

alshakero commented on client/my-sites/checkout/composite-checkout/payment-method-processors.js:
```diff
@@ -256,12 +256,9 @@ export async function payPalProcessor(
-		pathname: getThankYouUrl(),
-	} );
+
+	const successUrl = resolveUrl( window.location.href, getThankYouUrl() );
```
@sirbrillig this small yet significant change. Would really appreciate your approval.

sirbrillig replied:
Thanks for calling that out. It looks fine to me! I tested a PayPal purchase and the successUrl looks correct.

alshakero commented:
Thank you so much for the reviews @tjcafferkey and @sirbrillig!! I addressed all your feedback and tested with WP credits and paypal (with and without support session).

sirbrillig reviewed (approved):
From Shilling: 👍

sirbrillig commented:
In https://github.com/Automattic/wp-calypso/pull/46423 I moved the thank-you function to TypeScript to help prevent bugs in the future. Sorry to create conflicts! They should be easy to resolve though:

- The relative path imports are fixed.
- The thank-you function is basically the same, just moved to a different file.
- The tests (thank you for adding them!) are unchanged so you should be able to easily verify that these changes still work.

alshakero commented:
Addressed the conflicts, tested with PayPal (with Concierge session) and all works now.
